SvgBatchConverter: refactoring

simon04 opened the pull request:
Please see the individual commits for details.

kirill-grouchnikov commented on photon/src/main/java/org/pushingpixels/photon/api/transcoder/SvgBatchConverter.java:
```diff
@@ -32,96 +32,88 @@
-        File outputFolder = new File(outputFolderName);
-        if (!outputFolder.exists()) {
-            return;
+        String outputPackageName = getInputArgument(args, "outputPackageName", sourceFolderName);
```
This is not right. Output package name doesn't default to source folder name (which is a file path that may indeed contain the package name in it).

simon04 replied:
Thanks for spotting this glitch!

kirill-grouchnikov commented on photon/src/main/java/org/pushingpixels/photon/api/transcoder/SvgBatchConverter.java:
```diff
@@ -132,42 +124,53 @@ public static void main(String[] args) {
+                        .replace('-', '_')
+                        .replace(' ', '_');
+
+                // the image ${inputFolder}/foo/bar/baz.svg is written to ${inputFolder}/foo/bar/${outputClassNamePrefix}_baz.java
```
is written -> is transcoded
to ${inputFolder} -> to ${outputFolder}

kirill-grouchnikov replied:
Also the underscore is not automatically added, unlike this comment implies

kirill-grouchnikov commented on photon/src/main/java/org/pushingpixels/photon/api/transcoder/SvgBatchConverter.java:
```diff
@@ -132,42 +124,53 @@ public static void main(String[] args) {
+
+                // the image ${inputFolder}/foo/bar/baz.svg is written to ${inputFolder}/foo/bar/${outputClassNamePrefix}_baz.java
+                // the class ${outputClassNamePrefix}_baz is placed in package ${outputPackageName}.foo.bar
+                final Path relativeParent = inputFolder.relativize(path).getParent();
```
I'm not the biggest fan of how complex the logic is with this change. Before it's a rather straightforward iteration over all SVG files in a single folder. Now there's a lot of complexity in here for no apparent gain other than using stream / map APIs.

I'd rather keep the logic simple and have multiple invocations of the main converter in the script than push this complexity into the converter.

If you have something non-trivial in scope in mind, it would be better to first discuss that in bug tracker beforehand.

simon04 commented:
Thanks for your review. I addressed your comments.

Regarding the recursive transcoding: I'm testing Photon [on 929 SVG files](https://josm.openstreetmap.de/browser/josm/trunk/resources/images) of @JOSM with a nested structure. Defining a dozen separate invocations seems tedious and error prone to me. I'm happy to create a separate PR for this change if it requires more discussion.

kirill-grouchnikov commented:
Let's first discuss this at https://github.com/kirill-grouchnikov/radiance/issues/251

simon04 commented:
Alright. I've removed the last commit (for the recursive transcoding). Now, this PR contains the refactoring only.
